### PR TITLE
Replace usage of glamor CSSProperties with React CSSProperties

### PR DESCRIPTION
**Pull request by joel-jeremy**
Relates to #701

Gist of changes:
- Replace usage of glamor's `CSSProperties` with custom `CSSProperties` that extends from React's.
- Replace `style={[...]}` with `style={{...}}`
- Replace usage of `HTMLPropsWithStyle` (which was created because of glamor's style typing) with React's `HTMLProps`
  ~~- Still need to remove `HTMLPropsWithStyle` in  `View.tsx` but I will put that in a separate PR to make review easier.~~

### CODE REVIEW

File: packages/desktop-client/src/components/autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -590,6 +591,14 @@ function MultiAutocomplete({
     prevOnKeyDown?.(e);
   }
 
+  const inputStyle = {
+    flex: 1,
+    minWidth: 30,
+    border: 0,
+    ':focus': { border: 0, boxShadow: 'none' },
+    ...props?.inputProps?.style,
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.tsx:599]: Used `inputProps.style` since `props` doesn't have a `style` property

**Comment by MatissJanis** [comment]
Updated with latest changes in `master` to see if VRT still passes in this PR. Hope you don't mind.

**Comment by MatissJanis** [comment]
Looks like this introduces some visual regressions (see CI job). Would you mind taking a look?

**Comment by joel-jeremy** [comment]
VRT is failing in CI but when I run it locally all the tests are passing

File: packages/desktop-client/src/components/Notifications.tsx [hunk 2]
```diff
@@ -172,13 +177,7 @@ function Notification({
...
                 fontSize: 14,
                 flexShrink: 0,
-                '&:hover, &:active': {
-                  backgroundColor: positive
-                    ? colors.g9
-                    : error
-                    ? colors.r10
-                    : colors.y9,
-                },
+                ...buttonStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/Notifications.tsx:180]: **❓ question:** ‏is this change necessary? Functionally it's exactly the same

File: packages/desktop-client/src/components/budget/rollover/rollover-components.tsx [hunk 3]
```diff
@@ -551,7 +557,7 @@ export function IncomeCategoryMonth({
             type="financial"
             style={{
               cursor: 'pointer',
-              ':hover': { textDecoration: 'underline' },
+              ...hoverStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/rollover/rollover-components.tsx:560]: Same question here: I'm not quite understanding why we would need this change.

File: packages/desktop-client/src/components/table.tsx [hunk 4]
```diff
@@ -366,7 +365,7 @@ type InputCellProps = ComponentProps<typeof Cell> & {
   inputProps: ComponentProps<typeof InputValue>;
   onUpdate: ComponentProps<typeof InputValue>['onUpdate'];
   onBlur: ComponentProps<typeof InputValue>['onBlur'];
-  textAlign?: string;
+  textAlign?: CSSProperties['textAlign'];
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/table.tsx:368]: **👏 praise:** ‏nice!

File: packages/desktop-client/src/components/table.tsx [hunk 5]
```diff
@@ -648,10 +653,7 @@ export function SelectCell({
             backgroundColor: selected
               ? theme.tableTextEditingBackground
               : theme.tableBackground,
-            ':focus': {
-              border: '1px solid ' + theme.altFormInputBorderSelected,
-              boxShadow: '0 1px 2px ' + theme.altFormInputShadowSelected,
-            },
+            ...focusStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/table.tsx:656]: Same question here

File: packages/desktop-client/src/style/styles.ts [hunk 6]
```diff
@@ -103,31 +104,35 @@ let hiddenScrollbars = false;
...
-      borderRadius: 30,
-      backgroundClip: 'padding-box',
-      border: '2px solid rgba(0, 0, 0, 0)',
-    },
-    '& ::-webkit-scrollbar-thumb:vertical': {
-      backgroundColor: '#d0d0d0',
-    },
-  };
+  styles.lightScrollbar = {
+    ...(!hiddenScrollbars && {
```
Comment (MatissJanis) [inline packages/desktop-client/src/style/styles.ts:108]: **💬 suggestion:** ‏object spread is not necessary here

File: packages/desktop-client/src/style/styles.ts [hunk 7]
```diff
@@ -103,31 +104,35 @@ let hiddenScrollbars = false;
...
-    '& ::-webkit-scrollbar': {
-      width: 7,
-      backgroundColor: 'rgba(0, 0, 0, 0)',
-    },
-    '& ::-webkit-scrollbar-thumb:vertical': {
-      backgroundColor: 'rgba(200, 200, 200, .5)',
-    },
-  };
+  styles.darkScrollbar = {
+    ...(!hiddenScrollbars && {
```
Comment (MatissJanis) [inline packages/desktop-client/src/style/styles.ts:126]: Another unnecessary object spread

File: packages/desktop-client/src/components/budget/report/components.tsx [hunk 8]
```diff
@@ -349,9 +351,7 @@ export const CategoryMonth = memo(function CategoryMonth({
             getStyle={makeAmountGrey}
             style={{
               cursor: 'pointer',
-              ':hover': {
-                textDecoration: 'underline',
-              },
+              ...spentHoverStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/report/components.tsx:354]: Same question here

File: packages/desktop-client/src/components/budget/report/components.tsx [hunk 9]
```diff
@@ -297,20 +306,13 @@ export const CategoryMonth = memo(function CategoryMonth({
...
-                backgroundColor: 'white',
-              },
-            },
-          ]}
+          valueStyle={{
+            cursor: 'default',
+            margin: 1,
+            padding: '0 4px',
+            borderRadius: 4,
+            ...budgetHoverStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/report/components.tsx:314]: And here

File: packages/desktop-client/src/components/common/InputWithContent.tsx [hunk 10]
```diff
@@ -46,18 +57,11 @@ export default function InputWithContent({
...
-              backgroundColor: 'transparent',
-              boxShadow: 'none',
-              color: 'inherit',
-            },
-          },
-        ]}
+        style={{
+          ...inputStyle,
+          flex: 1,
+          ...hoverStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/common/InputWithContent.tsx:63]: same question

File: packages/desktop-client/src/components/common/Search.tsx [hunk 11]
```diff
@@ -65,19 +79,7 @@ export default function Search({
...
-        },
-        ':focus': isInModal
-          ? null
-          : {
-              '::placeholder': {
-                color: theme.formInputTextPlaceholderSelected,
-              },
-            },
-      }}
+      inputStyle={inputStyle}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/common/Search.tsx:82]: **❓ question:** ‏is this change necessary?

File: packages/desktop-client/src/components/settings/Global.tsx [hunk 12]
```diff
@@ -58,10 +65,7 @@ export default function GlobalSettings() {
           borderRadius: 4,
           overflow: 'auto',
           whiteSpace: 'nowrap',
-          // TODO: When we update electron, we should be able to
-          // remove this. In previous versions of Chrome, once the
-          // scrollbar appears it never goes away
-          '::-webkit-scrollbar': { display: 'none' },
+          ...webkitScrollbarStyle,
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/settings/Global.tsx:68]: **❓ question:** ‏is this change necessary?

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Code looks mostly good. Left a couple of questions and suggestions, but overall LGTM.

As for the VRT tests: I think the removal of `glamor` in some places has _slightly_ changed the font. Which then breaks VRT. But that's OK. I would recommend running VRT to update the screenshots and then commit them in this PR.

```
E2E_START_URL=https://my-remote-server.com yarn vrt --update-snapshots
```

File: packages/desktop-client/src/components/Notifications.tsx (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/Notifications.tsx:180]: I had to do these because React's  `CSSProperties` does not define these and results in a compiler error. This is a way to get around that. Do you have any suggestions how else we can address this? Does making our own `CSSProperties` which extends React's and add a `[propertyName: string]: unknown` property make more sense?

File: packages/desktop-client/src/style/styles.ts (hunk 6 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/style/styles.ts:108]: Same as my previous comment

File: packages/desktop-client/src/style/styles.ts (hunk 7 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/style/styles.ts:126]: Same as my previous comment

File: packages/desktop-client/src/components/Notifications.tsx (hunk 2 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/Notifications.tsx:183]: Yes, I think a custom type would be more sensible at this point. Reason being: we're still internally using `glamor` and not fully native React `style`. And we won't be able to port over to native until we solve all these issues. It will be easier to spot and solve them if a custom type is used rather than manual type coercions and workarounds.

**Comment by MatissJanis** [comment]
Judging by the screenshots - there is a small regression in settings page and also mobile budget page. Schedules page looks OK though (maybe like 1px difference or some font change, but nothing significant we should worry about).

**Comment by joel-jeremy** [comment]
This VRT is very cool! Gives us confidence in making style changes/refactoring. :) Looks like I have address all the other regressions.

File: packages/desktop-client/src/components/budget/MobileBudgetTable.js [hunk 13]
```diff
@@ -1058,29 +1067,26 @@ function BudgetHeader({
...
-        ]}
+        style={{
+          ...styles.mediumText,
+          marginTop: 12,
+          marginBottom: 12,
+          color: colors.n11,
+          textAlign: 'center',
+          // zIndex: -1
+        }}
+        data-vrt-mask
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/budget/MobileBudgetTable.js:1089]: **💬 suggestion:** ‏you'll want to remove this line and re-run the VRT tests. We don't need to mask dates anymore :)
```suggestion
```

File: packages/desktop-client/src/components/sidebar/ItemContent.tsx [hunk 14]
```diff
@@ -28,16 +25,14 @@ function ItemContent({
...
-          userDrag: 'none',
-          cursor: 'pointer',
-          ...(forceActive ? activeStyle : {}),
-        },
-      ]}
+      style={{
+        ...style,
+        touchAction: 'auto',
+        userSelect: 'none',
+        // userDrag: 'none',
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/ItemContent.tsx:32]: Intentional change?

File: packages/desktop-client/src/style/styles.ts (hunk 6 above)
Comment (MatissJanis) [reply packages/desktop-client/src/style/styles.ts:108]: I think with your latest type change we can now revert the changes in this file

File: packages/desktop-client/src/components/sidebar/ItemContent.tsx (hunk 14 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/sidebar/ItemContent.tsx:32]: This doesn't seem to be a valid css property and compiler is complaining so I commented it out. But compiler is no longer complaining so I'll revert :)

File: packages/desktop-client/src/components/budget/MobileBudgetTable.js (hunk 13 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/budget/MobileBudgetTable.js:1089]: Removed 👍

**Review by MatissJanis — APPROVED** [review]
Amazing! Thanks for all the hard work!
